### PR TITLE
List articles

### PR DESCRIPTION
**Pull request by JessSaavedra**
## Summary

- Listed Articles with AJAX
- Paginated with Kaminari

## Trello Card

https://trello.com/c/5xzVfs5N

### CODE REVIEW

File: app/assets/javascripts/pagination.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+       url: this.href,
+       dataType: "json" 
+    }).done(function(data) {
+      $("#articles").html(makeTable(data.articles));
+    })
+    return false; 
+  });
+})
+
+function makeTable(articles) {
```
Comment (epintos) [inline app/assets/javascripts/pagination.js]: Just for you to know, its better to define methods before you use them in Javascript. So this one should go above

File: app/controllers/api/v1/articles_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class Api::V1::ArticlesController < Api::V1::BaseController
+  def show
+    article = Article.find(params[:id])
+
+    render(json: ArticleSerializer.new(article).to_json)
```
Comment (epintos) [inline app/controllers/api/v1/articles_controller.rb]: `render json: article` uses the ArticleSerializer by default. Just for you to know

File: app/controllers/api/v1/articles_controller.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  def index
+    articles = Article.all.page(params[:page]).per(5)
+
+    render(
```
Comment (epintos) [inline app/controllers/api/v1/articles_controller.rb]: Same here

File: app/presenters/v1/base_presenter.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+class V1::BasePresenter
```
Comment (epintos) [inline app/presenters/v1/base_presenter.rb]: where do yo uuse this? Its good that you make your own code review as soon as you create the PR to detect unused files, little bugs and other things

File: app/views/articles/_articles.html.slim [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+table.table.table-striped
+  thead
+    tr
+      th= model_class.human_attribute_name(:id)
+      th= model_class.human_attribute_name(:title)
+      th= model_class.human_attribute_name(:text)
+      th= model_class.human_attribute_name(:created_at)
+      th= t '.actions', :default => t("helpers.actions")
+  tbody
+    - @articles.each do |article|
```
Comment (epintos) [inline app/views/articles/_articles.html.slim]: Remember not to use variables with `@` in partials

File: public/docs/v1/index.html [hunk 6]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
+<html lang="en-US">
```
Comment (epintos) [inline public/docs/v1/index.html]: This was autogerenerated right?

File: spec/requests/v1/base_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
```
Comment (epintos) [inline spec/requests/v1/base_controller_spec.rb]: Unused files

**Comment by epintos** [comment]
:+1:

**Comment by JessSaavedra** [comment]
Merged manually

**Pull request closed without merging**